subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -275,55 +321,87 @@ func (h *HivedAlgorithm) GetAffinityGroup(name string) api.AffinityGroup {
 		name)))
 }
 
+func (h *HivedAlgorithm) GetClusterStatus() api.ClusterStatus {
+	h.algorithmLock.RLock()
+	defer h.algorithmLock.RUnlock()
+
+	return h.clusterStatus
```

Comment:
>clusterStatus [](start = 10, length = 13)

clusterStatus is just a reference, after it is returned. It can be read outside, while internal still writing it?

Reply:
Have you confirmed previous code (return reference) really have issues?

---
In reply to: [388137116](https://github.com/microsoft/pai/pull/4153#discussion_r388137116) [](ancestors = 388137116)